Alignment/CommonAlignmentProducer/plugins/PCLTrackerAlProducer.cc

```diff
@@ -0,0 +1,1247 @@
+                              << "EventSetup-Record changed.";
+    initAlignmentAlgorithm(setup);
+  }
+
+  // Do not forward edm::Run
+  theAlignmentAlgo->beginRun(setup);
+
+  if (setupChanged(setup)) {
+    initAlignmentAlgorithm(setup);
+  }
+
+
+}
+
+//_____________________________________________________________________________
+void PCLTrackerAlProducer
+::endRun(const edm::Run& run, const edm::EventSetup& setup)
+{
+  // TODO: Either MP nor HIP is implementing the endRun() method... so this
+  //       seems to be useless?
+
+  if (tkLasBeamTag_.encode().size()) {
+    edm::Handle<TkFittedLasBeamCollection> lasBeams;
+    edm::Handle<TsosVectorCollection> tsoses;
+    run.getByLabel(tkLasBeamTag_, lasBeams);
+    run.getByLabel(tkLasBeamTag_, tsoses);
+
+    theAlignmentAlgo->endRun(EndRunInfo(run.id(), &(*lasBeams),
+                                        &(*tsoses)), setup);
+  } else {
+    edm::LogInfo("Alignment") << "@SUB=PCLTrackerAlProducer::endRun"
+                              << "No Tk LAS beams to forward to algorithm.";
+    theAlignmentAlgo->endRun(EndRunInfo(run.id(), 0, 0), setup);
+  }
+}
+
+//_____________________________________________________________________________
+void PCLTrackerAlProducer
+::beginLuminosityBlock(const edm::LuminosityBlock& lumiBlock,
+                       const edm::EventSetup&      setup)
+{
+  // Do not forward edm::LuminosityBlock
+  theAlignmentAlgo->beginLuminosityBlock(setup);
+}
+
+//_____________________________________________________________________________
+void PCLTrackerAlProducer
+::endLuminosityBlock(const edm::LuminosityBlock& lumiBlock,
+                     const edm::EventSetup&      setup)
+{
+  // Do not forward edm::LuminosityBlock
+  theAlignmentAlgo->endLuminosityBlock(setup);
+}
+
+//_____________________________________________________________________________
+void PCLTrackerAlProducer
+::analyze(const edm::Event& event, const edm::EventSetup& setup)
+{
+  if (!theAlignmentAlgo->processesEvents()) {
+    edm::LogWarning("BadConfig") << "@SUB=PCLTrackerAlProducer::analyze"
+                                 << "Skipping event. The current configuration "
+                                 << "of the alignment algorithm does not need "
+                                 << "to process any events.";
+    return;
+  }
+
+  if (setupChanged(setup)) {
+    edm::LogInfo("Alignment") << "@SUB=PCLTrackerAlProducer::analyze"
+                              << "EventSetup-Record changed.";
+    initAlignmentAlgorithm(setup);
+  }
+
+  if (nevent_== 0 && theExtraAlignables) {
+    initBeamSpot(event);
+  }
+
+  ++nevent_;
+
+  // reading in survey records
+  readInSurveyRcds(setup);
+
+  // Retrieve trajectories and tracks from the event
+  // -> merely skip if collection is empty
+  edm::Handle<TrajTrackAssociationCollection> handleTrajTracksCollection;
+
+  if (event.getByLabel(tjTkAssociationMapTag_, handleTrajTracksCollection)) {
```

Comment:
@cerminar As I see this is already done in another commit... is there a new PR for this commit?